File: JATmegaC/model/IntegerModel.cpp
/*
 * IntegerModel.cpp
 *
 *  Created on: 06-04-2011
 *      Author: radomir.mazon
 */

#include "IntegerModel.h"
#include <stdlib.h>


	IntegerModel::IntegerModel(){
		offset = 0;
	}

	char* IntegerModel::toString(char* result) {
		char temp[3] ={0};
		if (offset == 0) {
			result[0] = 0;
			return (char*)result;
		}

		if (offset >= 1) {
			itoa(tab[0], temp, 16);
			addZero(temp);
			result[0] = temp[0];
			result[1] = temp[1];
		}
		result[2] = 0;

		if (offset >= 2) {
			itoa(tab[1], temp, 16);
			addZero(temp);
			result[2] = temp[0];
			result[3] = temp[1];
		}
		result[4] = 0;

		if (offset >= 3) {
			itoa(tab[2], temp, 16);
			addZero(temp);
			result[4] = temp[0];
			result[5] = temp[1];
		}
		result[6] = 0;
		return result;
	}

	void IntegerModel::fromString(char* c) {
		int i=0;
		while(c[i] != 0) i++;
		offset = 2;
		char temp[3];
		temp[2] = 0;
		if (offset >=1) {
			temp[0] = c[0];
			temp[1] = c[1];
			tab[0] = atoi2(temp);
		}
		if (offset >=2) {
			temp[0] = c[2];
			temp[1] = c[3];
			tab[1] = atoi2(temp);
		}
		if (offset >=3) {
			temp[0] = c[4];
			temp[1] = c[5];
			tab[2] = atoi2(temp);
		}
	}

	void IntegerModel::setA(int a) {
		if (offset < 1) {
			offset = 1;
		}
		tab[0] = a;
	}

	void IntegerModel::setB(int b) {
		if (offset <2) {
			offset = 2;
		}
		tab[1] = b;
	}

	void IntegerModel::setC(int c) {
		offset = 3;
		tab[2] = c;
	}

	int IntegerModel::getA() {
		return tab[0];
	}

	int IntegerModel::getB() {
		return tab[1];
	}

	int IntegerModel::getC() {
		return tab[2];
	}

File: JATmegaJava/src/pl/mazon/jatmega/core/model/IntegerModel.java
package pl.mazon.jatmega.core.model;

public class IntegerModel implements IModel {

	private Integer[] data = new Integer[3];
	private int index =0;
	
	public Integer getA() {
		if (index < 1) {
			return null;
		}
		return data[0];
	}

	public void setA(Integer a) {
		if (index < 1) {
			index = 1;
		}
		this.data[0] = a;
	}

	public Integer getB() {
		if (index < 2) {
			return null;
		}
		return data[1];
	}

	public void setB(Integer b) {
		this.data[1] = b;
		if (index < 2) {
			index = 2;
		}	
	}

	public Integer getC() {
		if (index < 3) {
			return null;
		}
		return data[2];
	}

	public void setC(Integer c) {
		this.data[2] = c;
		if (index < 3) {
			index = 3;
		}
	}

	public void fromString(String from) {
		// FFFFFF
		if ( ! (from.length()%2 != 0) ) {
			throw new IllegalStateException("from="+from);
		}
		
		data[0] = Integer.parseInt(from.substring(0, 2), 16);
		index = 1;
		if (from.length() >=5) {
			data[1] = Integer.parseInt(from.substring(2, 4), 16);
			index = 2;
		} 
		if (from.length() ==8) {
			data[2] = Integer.parseInt(from.substring(4, 6), 16);
			index = 3;
		}
	}
	
	@Override
	public String toString() {
		String d1 = "";
		String d2 = "";
		String d3 = "";
		if (index >=1) {
			d1 = Integer.toHexString(data[0]);
		}
		if (index >=2) {
			d2 = Integer.toHexString(data[1]);
		}
		if (index >=3) {
			d3 = Integer.toHexString(data[2]);
		}
		String result= d1 + d2 + d3;
		return result.toUpperCase();
	}
}
